### ISSUE TITLE
Support for bare git repositories

### ISSUE THREAD

**Issue by cstjean** [JuliaLang/Pkg.jl#659]
I'm updating my company's package to 0.7. So far, we've shared our changes in a [bare git repository](http://www.saintsjd.com/2011/01/what-is-a-bare-git-repository/), which is accessible in a shared folder over the network. In 0.7, I thought that I could `add /shared/MyPackage.git`, and indeed that works. Unfortunately, `using MyPackage` tells me that it "does not seem to be installed" and recommends `Pkg.instantiate()`. `instantiate` does not seem to do anything, and `st` still shows a red arrow next to my package.

My understanding is that `add` did not clone the repo. It's assuming incorrectly that `/shared/MyPackage.git` has a working tree, so of course it cannot find `/shared/MyPackage.git/src/MyPackage.jl`.

Perhaps `Pkg` should automatically clone bare git repositories? Or, if that's a bad setup altogether, it should detect it and give a more helpful error message.

**Comment by KristofferC** [comment]
> Or, if that's a bad setup altogether, it should detect it and give a more helpful error message.

Yes, it should verify the tree structure. I think it does that when there is a Project file...?

**Comment by cstjean** [comment]
If I have a registry with a package `repo` pointing to a bare git repo, that should be fine, right? I get `ERROR: MyPackage: git object a9a6dbcbb364b9309eeb0a977969d4876de6345a should be a tree,not LibGit2.GitCommit`

**Comment by KristofferC** [comment]
Sounds like you put a git commit instead of the tree sha in the registry?

**Comment by cstjean** [comment]
That was it, thank you.

**Comment by filchristou** [comment]
I still get the error from as described from @cossio.
Are there any workarounds or guidelines to get it to work?
